src/common/include/sirf/common/jcg.h

```diff
@@ -0,0 +1,123 @@
+#pragma once
+
+#include <cmath>
+#include <iostream>
+
+#include <boost/shared_ptr.hpp>
+
+template<class vector>
```

Comment:
maybe confusing to use `vector` (at first glance anyway). what about calling it `vector_type` or `vectorT` or something

Reply:
done